src/components/LmsConfigurations/MoodleIntegrationConfigForm.jsx

```diff
@@ -0,0 +1,339 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import { ValidationFormGroup, Input, StatefulButton, Icon } from '@edx/paragon';
+import { snakeCaseFormData } from '../../utils';
+import LmsApiService from '../../data/services/LmsApiService';
+import StatusAlert from '../StatusAlert';
+import NewRelicService from '../../data/services/NewRelicService';
+
+export const REQUIRED_MOODLE_CONFIG_FIELDS = [
+  'moodleBaseUrl',
+  'serviceShortName',
+];
+
+const SUBMIT_STATES = {
+  ERROR: 'error',
+  DEFAULT: 'default',
+  COMPLETE: 'complete',
+  PENDING: 'pending',
+};
+
+
+class MoodleIntegrationConfigForm extends React.Component {
+  state = {
+    invalidFields: {},
+    submitState: SUBMIT_STATES.DEFAULT,
+    active: this.props.config?.active,
+    error: undefined,
+  }
+
+  handleErrors(error) {
+    const errorMsg = error.message || error.response?.status === 500 ?
+      error.message : JSON.stringify(error.response.data);
+    NewRelicService.logAPIErrorResponse(errorMsg);
+    return errorMsg;
+  }
+
+    /**
+   * Creates a new third party provider configuration, then updates this list with the response.
+   * Returns if there is an error.
+   * @param {FormData} formData
+   */
+  createMoodleConfig = async (formData) => {
+    const transformedData = snakeCaseFormData(formData);
+    transformedData.append('enterprise_customer', this.props.enterpriseId);
+    try {
+      const response = await LmsApiService.postNewMoodleConfig(transformedData);
+      this.setState({ config: response.data });
+      return undefined;
```

Comment:
nit: did this in my pr, think we can just drop the return undefined and just use return; here

Reply:
Ah I keep getting this: `  48:51  error  Expected to return a value at the end of async arrow function  consistent-return` when attempting to do that or to just not have a return at all. :/